cocos/renderer/CCPass.cpp

```diff
@@ -247,33 +219,28 @@ void Pass::draw(float globalZOrder, backend::Buffer* vertexBuffer, backend::Buff
         _programState->setUniform(_locNormalMatrix, normalMatrix.data(), sizeof(normalMatrix[0]) * normalMatrix.size());
     }
 
-    auto *renderer = Director::getInstance()->getRenderer();
-
-    renderer->addCommand(&_meshCommand);
 }
 
-void Pass::onBeforeVisitCmd()
+void Pass::onBeforeVisitCmd(MeshCommand *command)
 {
     auto *renderer = Director::getInstance()->getRenderer();
-    //_oldDepthEnabledState = renderer->getDepthTest();
-
-    auto &pipelineDescriptor = _meshCommand.getPipelineDescriptor();
 
     _rendererDepthTestEnabled = renderer->getDepthTest();
     _rendererDepthCmpFunc = renderer->getDepthCompareFunction();
     _rendererCullMode = renderer->getCullMode();
     
     _rendererDepthWrite = renderer->getDepthWrite();
     _rendererWinding = renderer->getWinding();
-
-    _renderState.bindPass(this);
     renderer->setDepthTest(true);
+    _renderState.bindPass(this, command);
+
+    updateMVPUniform(command->getMV());
```

Comment:
update transform matrix in "before callback"

Reply:
So the `draw` function doesn't have to pass `modelView`.

Reply:
`modelView` is attached to meshCommand 
``` cpp
meshCommand->init(globalZOrder, modelView);
```